roles/edpm_derive_pci_device_spec/tasks/derive_pci_device_spec.yml

```diff
@@ -0,0 +1,51 @@
+---
+# Copyright 2024 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+- name: Check valid input for edpm_derive_sriov_device_spec_list
+  ansible.builtin.fail:
+    msg: "List of sriov device_spec cannot be empty - edpm_derive_sriov_device_spec_list"
+  when: edpm_derive_sriov_device_spec_list | length == 0
+
+- name: Create neutron-sriov-agent directories
+  become: true
+  ansible.builtin.file:
+    path: "{{ edpm_derive_pci_device_spec_conf_dir }}"
+    setype: "container_file_t"
+    state: directory
+    owner: "{{ ansible_user }}"
+    group: "{{ ansible_user }}"
+    mode: "0755"
+
+- name: Check if edpm node has the sriov nova conf
+  ansible.builtin.stat:
+    path: "{{ edpm_derive_pci_device_spec_conf_dir }}/{{ edpm_derive_pci_device_spec_conf_file }}"
+  register: sriov_device_spec_conf_check
+
+- name: Derive the sriov nic-partitioned pci device_spec addr
+  when:
+    - not sriov_device_spec_conf_check.stat.exists
+  become: true
+  block:
+    - name: Run derive_pci_passthrough_devicespec
+      edpm_derive_pci_passthrough_devicespec:
+        sriov_device_spec_map: "{{ edpm_derive_sriov_device_spec_list }}"
+        sriov_phydev_map: "{{ edpm_derive_pci_device_spec_physmap_file }}"
+        sriov_device_spec_out_file: "{{ edpm_derive_pci_device_spec_conf_dir }}/{{ edpm_derive_pci_device_spec_conf_file }}"
+      register: derive_pci_devicespec
+      changed_when: derive_pci_devicespec.rc == 0
+    - name: Print result of derived device_spec 
+      debug:
+        msg: "{{ derive_pci_devicespec }}"
```

Comment:
@vcandapp  Please use derive_pci_devicespec.device_spec instead of complete details

Reply:
Updated debug msg